base_layer/core/src/block.rs

```diff
@@ -157,4 +166,135 @@ impl From<Transaction> for AggregateBody {
     }
 }
 
+#[derive(Default)]
+pub struct BlockBuilder {
+    pub header: Option<BlockHeader>,
+    pub inputs: Option<Vec<TransactionInput>>,
+    pub outputs: Option<Vec<TransactionOutput>>,
+    pub kernels: Option<Vec<TransactionKernel>>,
+}
+
+impl BlockBuilder {
+    pub fn new() -> BlockBuilder {
+        BlockBuilder {
+            header: None,
+            inputs: None,
+            outputs: None,
+            kernels: None,
+        }
+    }
+
+    /// This function adds a header to the block
+    pub fn with_header(&mut self, header: BlockHeader) -> &Self {
```

Comment:
This function seems somewhat pointless. It feel more ergonomic to take an Option(previous block) as an argument to the builder constructor and then automatically generate the header from that info. Passing None means genesis block.

Reply:
The problem of constructing a block from the previous block is the MMR roots, you need the whole utxo set for that. I feel its better to completely leave that out because of that.